lms/static/js/spec/verify_student/image_input_spec.js

```diff
@@ -0,0 +1,157 @@
+define([
+    'jquery',
+    'backbone',
+    'js/common_helpers/template_helpers',
+    'js/common_helpers/ajax_helpers',
+    'js/verify_student/views/image_input_view',
+    'js/verify_student/models/verification_model'
+], function( $, Backbone, TemplateHelpers, AjaxHelpers, ImageInputView, VerificationModel ) {
+    'use strict';
+
+    describe( 'edx.verify_student.ImageInputView', function() {
+
+        var IMAGE_DATA = 'abcd1234';
+
+        var createView = function() {
+            return new ImageInputView({
+                el: $( '#current-step-container' ),
+                model: new VerificationModel({}),
+                modelAttribute: 'faceImage',
+                errorModel: new ( Backbone.Model.extend({}) )(),
+                submitButton: $( '#submit_button' ),
+            }).render();
+        };
+
+        var uploadImage = function( view, fileType, callback ) {
+            var imageCapturedEvent = false,
+                errorEvent = false;
+
+            // Since image upload is an asynchronous process,
+            // we need to wait for the upload to complete
+            // before checking the outcome.
+            runs(function() {
+                var fakeFile,
+                    fakeEvent = { target: { files: [] } };
+
+                // If no file type is specified, don't add any files.
+                // This simulates what happens when the user clicks
+                // "cancel" after clicking the input.
+                if ( fileType !== null) {
+                    fakeFile = new Blob(
+                        [ IMAGE_DATA ],
+                        { type: 'image/' + fileType }
+                    );
+                    fakeEvent.target.files = [ fakeFile ];
+                }
+
+                // Wait for either a successful upload or an error
+                view.on( 'imageCaptured', function() {
+                    imageCapturedEvent = true;
+                });
+                view.on( 'error', function() {
+                    errorEvent = true;
+                });
+
+                // Trigger the file input change
+                // It's impossible to trigger this directly due
+                // to browser security restrictions, so we call
+                // the handler instead.
+                view.handleInputChange( fakeEvent );
+            });
+
+            // Check that the image upload has completed,
+            // either successfully or with an error.
+            waitsFor(function() {
+                return ( imageCapturedEvent || errorEvent );
+            });
+
+            // Execute the callback to check expectations.
+            runs( callback );
+        };
+
+        var expectPreview = function( view, fileType ) {
+            var previewImage = view.$preview.attr('src');
+            if ( fileType ) {
+                expect( previewImage ).toContain( 'data:image/' + fileType );
+            }
+            else {
```

Comment:
nitpick: JavaScript formatting for if else should have the closing bracket and else on the same line